### PR TITLE
Use VPA CRD and coredns HelmReleases from cluster chart

### PR DESCRIPTION
**Pull request by nprokopic**
Also use default repositories from cluster chart.

Towards https://github.com/giantswarm/roadmap/issues/2851

### What this PR does / why we need it

- Use default HelmRepositories from cluster chart.
- Use vertical-pod-autoscaler-crd HelmRelease from cluster chart.
- Use coredns HelmRelease from cluster chart.
- Remove default HelmRepositories from cluster-aws.
- Remove vertical-pod-autoscaler-crd HelmRelease from cluster-aws.
- Remove coredns HelmRelease from cluster-aws.

### Checklist

- [x] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

**Comment by nprokopic** [comment]
This is the only significant change - vertical-pod-autoscaler-crd got bumped to v3, which should be fine, as there is already a PR in cluster-aws for that https://github.com/giantswarm/cluster-aws/pull/484.

```
/spec/chart/spec/version  (HelmRelease/org-giantswarm/test-mc-proxy-vertical-pod-autoscaler-crd)
  ± value change
    - 2.0.1
    + 3.0.0
```

**Comment by nprokopic** [comment]
Standard and Upgrade Suites passed, Private suite failed with

```
Common tests basic [It] has all of its Pods in the Running state
/app/internal/common/basic.go:95

  [FAILED] Timed out after 900.779s.
  Expected success, but got an error:
      <*errors.errorString | 0xc00055d620>: 
      pod kube-system/promtail-hfvtf in Pending phase
      {
          s: "pod kube-system/promtail-hfvtf in Pending phase",
      }
  In [It] at: /app/internal/common/basic.go:99 @ 02/22/24 00:42:27.591
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## [Unreleased]
 
+### Changed
+
+- Use default HelmRepositories from cluster chart.
+- Use vertical-pod-autoscaler-crd HelmRelease from cluster chart.
+- Use coredns HelmRelease from cluster chart.
+- Remove default HelmRepositories from cluster-aws.
+- Remove vertical-pod-autoscaler-crd HelmRelease from cluster-aws.
+- Remove coredns HelmRelease from cluster-aws.
```
Comment (AndiDog) [inline CHANGELOG.md:17]: As a general comment (not urgent in `v0.y.z`): the changelog should answer "what changes for customers"? For example in this case, it should state that essentially nothing changes.

**Review by AndiDog — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (nprokopic) [reply CHANGELOG.md:17]: Oh, good point.

**Pull request merged**